### DISCUSSION TITLE
[CONTRIBUTION] Speech Dataset Generator

### DISCUSSION THREAD

**Discussion by davidmartinrius** [enhuiz/vall-e#117] [Show and tell]
Hi everyone! 

I have just published this project on GitHub: https://github.com/davidmartinrius/speech-dataset-generator/

Now you can create datasets automatically with any audio or lists of audios.

I hope you can find it useful. 

## Here are the key functionalities of the project:

1. **Dataset Generation:** Creation of multilingual datasets with Mean Opinion Score (MOS).

2. **Silence Removal:** It includes a feature to remove silences from audio files, enhancing the overall quality.

3. **Sound Quality Improvement:** It improves the quality of the audio when needed.

4. **Audio Segmentation:** It can segment audio files within specified second ranges.

5. **Transcription:** The project transcribes the segmented audio, providing a textual representation.

6. **Gender Identification:** It identifies the gender of each speaker in the audio.

7. **Pyannote Embeddings:** Utilizes pyannote embeddings for speaker detection across multiple audio files.

8. **Automatic Speaker Naming:** Automatically assigns names to speakers detected in multiple audios.

9. **Multiple Speaker Detection:** Capable of detecting multiple speakers within each audio file.

10. **Store speaker embeddings:** The speakers are detected and stored in a Chroma database, so you do not need to assign a speaker name.

11. **Syllabic and words-per-minute metrics**

12. **Multiple input sources:** You can either use your own files or download content by pasting URLs from sources such as **YouTube**, **LibriVox** and **TED Talks**. 

Feel free to explore the project at https://github.com/davidmartinrius/speech-dataset-generator

**Comment by hoowee** [comment]
您好，您的邮件我已经收到，会尽快处理，谢谢！中建八局设计管理总院 胡伟

**Reply by davidmartinrius** [reply]
what email? I think I have never contacted you
